gammapy/data/observations.py

```diff
@@ -6,17 +6,57 @@
 from astropy.coordinates import SkyCoord
 from astropy.time import Time
 from astropy.units import Quantity
+from astropy.io import fits
+from astropy.table import Table
+from gammapy.utils.scripts import make_path
 from gammapy.utils.fits import LazyFitsData, earth_location_from_dict
 from gammapy.utils.testing import Checker
-from .event_list import EventListChecker
+from .event_list import EventList, EventListChecker
 from .filters import ObservationFilter
 from .gti import GTI
 from .pointing import FixedPointingInfo
+from .hdu_index_table import HDUIndexTable
+from gammapy import irf
 
 __all__ = ["Observation", "Observations"]
 
 log = logging.getLogger(__name__)
 
+irf_dictionary = HDUIndexTable.HDU_CLASS_IMPLEMENTATION
+
+def read_irf_with_hdu_class(irf_file, hdu_class):
```

Comment:
Here is one more idea: `Map.read()` acts as a "factory method" for maps. This means `Map.read()` will try to guess the map type from the file and return the correct instance off the class, such as `WcsNDMap`, `HpxNDMap` or `RegionNDMap`. It might be convenient to introduce the same system for `IRF.read()`...